### ISSUE TITLE
Nesting two identical loops is not working

### ISSUE THREAD

**Issue by roadster31** [thelia/thelia#2213]
Nesting two identical loops (except name) is not working: the outer loop runs only once.

Example:
```
<ul>
{loop type="product" name="out"}
    <li>{$REF}: {$TITLE}
        <ul>
            {loop type="product" name="in"}
                <li>{$REF}: {$TITLE}</li>
            {/loop}
        </ul>
    </li>
{/loop}
</ul>
```
Gives :
<ul>
    <li>1_2_75149428: en_US:Consectetur.
        <ul>
                <li>1_2_75149428: en_US:Consectetur.</li>
                <li>4_0_79316503: en_US:Magnam aut unde a.</li>
                <li>3_1_6937063: en_US:Maiores voluptas.</li>
                <li>2_0_33470529: en_US:Non necessitatibus.</li>
                <li>2_1_71394416: en_US:Sint voluptatem et.</li>
                <li>1_1_66884548: en_US:Sit ipsam.</li>
                <li>4_1_6623730: en_US:Velit et eos ut.</li>
        </ul>
    </li>
</ul>

Adding a parameter to one of the loops like solves the problem :
```
<ul>
{loop type="product" name="out"}
    <li>{$REF}: {$TITLE}
        <ul>
            {loop type="product" name="in" order="ref"}
                <li>{$REF}: {$TITLE}</li>
            {/loop}
        </ul>
    </li>
{/loop}
</ul>
```

**Comment by bibich** [comment]
maybe an issue with cache system of the loops. 
The results of a loop are cached. The results of a second loop with the same arguments (except name) are retrieved from cache.
The argument `no-cache` , available for every loops, disables this behaviour : ` no-cache="on" `

**Comment by roadster31** [comment]
using no-cache is not a valid solution. Loop arguments may be dynamically set, and a situation where two loops with the same argument set may occur behind the back of the developper.

Cloning the loopResult object returned by the cache is a simple and robust solution.

**Comment by bibich** [comment]
closed by #2214
